src/main/webapp/dev/js/common/questionMsq.es6

```diff
@@ -2,6 +2,131 @@ import {
     ParamsNames,
 } from './const.es6';
 
+function isMaxSelectableChoicesEnabled(questionNum) {
+    return $(`#msqEnableMaxSelectableChoices-${questionNum}`).prop('checked');
+}
+
+function isMinSelectableChoicesEnabled(questionNum) {
+    return $(`#msqEnableMinSelectableChoices-${questionNum}`).prop('checked');
+}
+
+function isGenerateOptionsEnabled(questionNum) {
+    return $(`#generateMsqOptionsCheckbox-${questionNum}`).prop('checked');
+}
+
+function getNumOfMsqOptions(questionNum) {
+    return $(`#msqChoiceTable-${questionNum}`).children().length - 1;
+}
+
+function getMaxSelectableChoicesElement(questionNum) {
+    return $(`#msqMaxSelectableChoices-${questionNum}`);
+}
+
+function getMaxSelectableChoicesValue(questionNum) {
+    if (isMaxSelectableChoicesEnabled(questionNum)) {
+        return parseInt(getMaxSelectableChoicesElement(questionNum).val(), 10);
+    }
+
+    // return infinity
+    return Number.MAX_SAFE_INTEGER;
+}
+
+function setUpperLimitForMaxSelectableChoices(questionNum, upperLimit) {
+    getMaxSelectableChoicesElement(questionNum).prop('max', upperLimit);
+}
+
+function setMaxSelectableChoices(questionNum, newVal) {
+    if (newVal >= 2) {
+        // No use if max selectable choices were 1
+        getMaxSelectableChoicesElement(questionNum).val(newVal);
+    }
+}
+
+function getMinSelectableChoicesElement(questionNum) {
+    return $(`#msqMinSelectableChoices-${questionNum}`);
+}
+
+function getMinSelectableChoicesValue(questionNum) {
+    if (isMinSelectableChoicesEnabled(questionNum)) {
+        return parseInt(getMinSelectableChoicesElement(questionNum).val(), 10);
+    }
+
+    // return infinity
+    return Number.MAX_SAFE_INTEGER;
+}
+
+function setMinSelectableChoices(questionNum, newVal) {
+    if (newVal >= 1) {
+        // No use if min selectable choices where 0
+        getMinSelectableChoicesElement(questionNum).val(newVal);
+    }
+}
+
+function setUpperLimitForMinSelectableChoices(questionNum, upperLimit) {
+    getMinSelectableChoicesElement(questionNum).prop('max', upperLimit);
+}
+
+/**
+ * Returns total number of options for the selected generate options type.
+ * Eg. if 'instructors' is selected, returns number of instructors for feedback session.
+ * Assumes that 'generateOptions' checkbox is checked.
+ */
+function getTotalOptionsForSelectedGenerateOptionsType(questionNum) {
+    let val;
+    const category = $(`#msqGenerateForSelect-${questionNum}`).prop('value');
+
+    if (category === 'STUDENTS') {
+        val = $('#num-students').val();
+    } else if (category === 'TEAMS') {
+        val = $('#num-teams').val();
+    } else {
+        val = $('#num-instructors').val();
+    }
+
+    return val;
```

Comment:
Rather than doing `let val` and `return val` at the end, just do:
```js
if () {
    return ...;
} else if () {
    return ...;
}
return ...;
```

Reply:
Isn't single exit point for a method considered to be better code design?

Reply:
You can't really justify why it is "considered to be better code design"; there are as many cons as there are pros to both sides. This is a subject that is [heavily debated](https://softwareengineering.stackexchange.com/questions/118703/where-did-the-notion-of-one-return-only-come-from), much like the tab-vs-space debate.

It boils down somewhat to personal preference; for me the multiple `return`s are clearer, so I'll give you the liberty to decide.